x-pack/plugins/security_solution/scripts/endpoint/common/endpoint_metadata_services.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { KbnClient } from '@kbn/test';
+import { WriteResponseBase } from '@elastic/elasticsearch/lib/api/types';
+import { clone, merge } from 'lodash';
+import { DeepPartial } from 'utility-types';
+import { resolvePathVariables } from '../../../public/common/utils/resolve_path_variables';
+import { HOST_METADATA_GET_ROUTE, METADATA_DATASTREAM } from '../../../common/endpoint/constants';
+import { HostInfo, HostMetadata } from '../../../common/endpoint/types';
+import { EndpointDocGenerator } from '../../../common/endpoint/generate_data';
+import { checkInFleetAgent } from './fleet_services';
```

Comment:
Some of these need to be imported as `type`. I'll go ahead and create a PR to add the missing rule to our eslint.